manila/share/drivers/scality/driver.py

```diff
@@ -45,44 +72,157 @@
+                login=self.configuration.management_user,
+                privatekey=self.configuration.ssh_key_path,
+                max_size=1)
+
+            export_ip = self.configuration.get(proto + '_export_ip')
+
+            if protocol == 'NFS':
+                self._helpers['NFS'] = NFSHelper(ssh_pool, export_ip)
+            else:
+                export_root = self.configuration.smb_export_root
+                # Manila refers to this protocol as CIFS.
+                # It's better known as SMB at Scality
+                self._helpers['CIFS'] = CIFSHelper(ssh_pool, export_ip,
+                                                   export_root)
+
+    def _get_helper(self, share):
+        """Get the correct helper instance based on the share protocol."""
+        helper = self._helpers.get(share['share_proto'])
+        if helper:
+            return helper
+        else:
+            reason = _("Protocol '%s' is wrong, unsupported or "
+                       "disabled") % share['share_proto']
+            raise exception.InvalidShare(reason=reason)
+
+    def check_for_setup_error(self):
+        """Returns an error if prerequisites aren't met.
+
+        This is called by `manila.share.manager.ShareManager` right after the
+        call to `do_setup`
+        """
+        for proto, helper in list(self._helpers.items()):
+            if not helper.setup_is_valid():
+                self._helpers.pop(proto)
+
+        if not self._helpers:
+            msg = _("ScalityShareDriver is not properly initialized")
+            raise exception.ManilaException(msg)
+
+    def allow_access(self, context, share, access, share_server=None):
+        # NOTE(vponomaryov): use direct verification for case some additional
+        # level is added.
+        access_level = access['access_level']
+        if access_level not in (const.ACCESS_LEVEL_RW, const.ACCESS_LEVEL_RO):
+            raise exception.InvalidShareAccessLevel(level=access_level)
+
+        self._get_helper(share).allow_access(share, access)
+
+    def deny_access(self, context, share, access, share_server=None):
+        self._get_helper(share).deny_access(share, access)
+
+    def delete_share(self, context, share, share_server=None):
+        self._get_helper(share).delete_share(share)
+
+    def create_share(self, context, share, share_server=None):
+        self._get_helper(share).create_share(share)
+
+    def ensure_share(self, context, share, share_server=None):
+        self._get_helper(share).ensure_share(share)
+
+    def _update_share_stats(self):
+        backend_name = self.configuration.safe_get(
+            'share_backend_name') or 'Scality Ring Driver'
+
+        if 'NFS' in self._helpers and 'CIFS' in self._helpers:
+            protocol = 'NFS_CIFS'
+        elif 'NFS' in self._helpers:
+            protocol = 'NFS'
+        elif 'CIFS' in self._helpers:
+            protocol = 'CIFS'
```

Comment:
From what I can tell, (even if I DO find this to be strange behavior) the scheduler does not take protocol into account.